### PR TITLE
INT-3416 Add Needed Fields for Custom Store

### PR DESCRIPTION
**Pull request by justin-shipstation**
Adding support for some of the fields required to export existing Custom Stores into Connect.

https://auctane.atlassian.net/browse/INT-3416

### CODE REVIEW

File: src/public/orders/methods.ts [hunk 1]
```diff
@@ -25,4 +25,4 @@ export type ShipmentCreated = (transaction: Transaction, shipment: SalesOrderShi
  * Called when an order has been imported into a marketplace.
  */
 export type AcknowledgeOrders = (transaction: Transaction, notifications: SalesOrderNotification[])
-=> AcknowledgedSalesOrder[] | Promise<AcknowledgedSalesOrder[]>;
\ No newline at end of file
+=> AcknowledgedSalesOrder[] | Promise<AcknowledgedSalesOrder[]>;
```
Comment (lzhang) [inline src/public/orders/methods.ts:28]: So good

File: src/internal/orders/requested-fulfillment.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import Joi = require("joi");
+import { RequestedFulfillmentExtensions, RequestedFulfillmentPOJO } from "../../public/orders/requested-fulfillment";
+import { _internal } from "../common";
+import { AddressWithContactInfo } from "../common/addresses/address-with-contact-info";
+import { SalesOrderItem } from "./sales-order-item";
+import { ShippingPreferences } from "./shipping-preferences";
+
+export class RequestedFulfillment {
```
Comment (jeversmann-shipstation) [inline src/internal/orders/requested-fulfillment.ts:8]: Thanks for giving this a type name 👍

File: src/public/orders/get-sales-orders-by-date-request.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+     * A customer specified desired mapping for status codes.
+     */
+    salesOrderStatusMapping?: SalesOrderStatusMapping;
+}
+
+export interface SalesOrderStatusMapping {
+    /**
+     * The raw status string used by the order source.
+     */
+    source_status: string;
```
Comment (jeversmann-shipstation) [inline src/public/orders/get-sales-orders-by-date-request.ts:18]: 🐫

Comment (justin-shipstation) [reply src/public/orders/get-sales-orders-by-date-request.ts:18]: oOps

File: src/public/orders/methods.ts (hunk 1 above)
Comment (justin-shipstation) [reply src/public/orders/methods.ts:28]: Doing the lords work, one new line at a time.

**Review by jeversmann-shipstation — APPROVED** [review]

**Comment by JamesMessinger** [comment]
This PR should not have been merged without approval from someone on the DX team.  There were major breaking changes made to the `GetSalesOrdersByDate` method, and some of these changes are not inline with existing conventions in the SDK.

Any change like this requires discussion and design thinking, without just jumping straight to coding — especially breaking changes.  In addition, documentation needs to be updated and version support needs to be considered.

**Pull request merged**